stytra/stimulation/__init__.py

```diff
@@ -109,10 +113,19 @@ def update_protocol(self):
         self.stimuli = self.protocol._get_stimulus_list()
 
         self.current_stimulus = self.stimuli[0]
+        
+        if hasattr(self.experiment, 'estimator'):
+            self.environment_state.estimator = self.experiment.estimator
 
         # pass experiment to stimuli for calibrator and asset folders:
         for stimulus in self.stimuli:
-            stimulus.initialise_external(self.experiment)
+            try:
+                stimulus.initialise_external(self.experiment, self.environment_state,)
+            except TypeError as e:
+                print("Error: {}".format(e))
+                stimulus.initialise_external(self.experiment)
+                warnings.warn("Warning: 'initialise_external' will use the environment_state variable which holds the calibrator and the estimator object!", FutureWarning)
+                warnings.warn("Warning: 'initialise_external' will use the environment_state variable which holds the calibrator and the estimator object!", DeprecationWarning)
```

Comment:
Tries to  call the updated function, if it can't it will only pass the experiment object -> further handling is done inside the function